OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java

```diff
@@ -1578,6 +1572,82 @@ private void parseLocation(SearchPhrase phrase, SearchResultMatcher resultMatche
 			}
 		}
 
+		private void sortCities(List<SearchResult> cities, final String phraseName) {
```

Comment:
Perhaps better to filtering city in publish method of SearchRequest<Amenity> req = BinaryMapIndexReader.buildSearchPoiRequest
Do not make sorting after